src/api/procedures/consumeJoinOrRotateAuthorization.ts

```diff
@@ -90,6 +94,12 @@ export async function prepareConsumeJoinOrRotateAuthorization(
       paidForBy: issuer,
       args: [rawAuthId],
     });
+  } else if (type === AuthorizationType.RotatePrimaryKey) {
+    this.addTransaction({
+      transaction: identity.acceptPrimaryKey,
+      paidForBy: issuer,
+      args: [rawAuthId, null],
+    });
   } else {
```

Comment:
These last two are very alike, we could avoid repeating the entire addTransaction call and only differentiate the value of `transaction` with a ternary